### PR TITLE
Maxim drivers for MAX32650 and MAX32655

### PR DESCRIPTION
**Pull request by CiprianRegus**
Updated the Maxim drivers to run on MAX32655 and MAX32650.

### CODE REVIEW

File: drivers/platform/maxim/maxim_hal.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#if TARGET_NUM == 32650
```
Comment (amiclaus) [inline drivers/platform/maxim/maxim_hal.h:1]: missing header file comments + ifndefs

File: drivers/platform/maxim/maxim_spi.c [hunk 2]
```diff
@@ -77,51 +78,67 @@ int32_t max_spi_init(struct no_os_spi_desc **desc,
 		return -ENOMEM;
 
 	eparam = param->extra;
+
```
Comment (amiclaus) [inline drivers/platform/maxim/maxim_spi.c:81]: nit: redundant new line

File: drivers/platform/maxim/maxim_spi.c [hunk 3]
```diff
@@ -77,51 +78,67 @@ int32_t max_spi_init(struct no_os_spi_desc **desc,
...
-		ret = MXC_SPI_Init(MXC_SPI1, SPI_MASTER_MODE, SPI_SINGLE_MODE,
-				   eparam->numSlaves, eparam->polarity, param->max_speed_hz);
-#endif
-	else {
+	if (descriptor->device_id >= MXC_SPI_INSTANCES) {
 		ret = -EINVAL;
 		goto err;
 	}
+		
+#if TARGET_NUM == 32655
```
Comment (amiclaus) [inline drivers/platform/maxim/maxim_spi.c:94]: what is `32655`? A relevant macro might help users/reviewers understand.

File: drivers/platform/maxim/spi_extra.h [hunk 4]
```diff
@@ -40,6 +40,8 @@
 #ifndef SPI_EXTRA_H_
 #define SPI_EXTRA_H_
 
+#include <stdint.h>
```
Comment (amiclaus) [inline drivers/platform/maxim/spi_extra.h:43]: is this required in the header file?

File: drivers/platform/maxim/irq_extra.h [hunk 5]
```diff
@@ -39,21 +39,9 @@
 #ifndef IRQ_MAXIM_EXTRA_H
 #define IRQ_MAXIM_EXTRA_H
 
-#include "max32660.h"
```
Comment (amiclaus) [inline drivers/platform/maxim/irq_extra.h:42]: are these two changes related? (including max_hal and removing irq_id)? if not, separate commits should be added.

**Review by amiclaus — COMMENTED** [review]
some comments on my side.

File: drivers/platform/maxim/maxim_spi.c (hunk 3 above)
Comment (CiprianRegus) [reply drivers/platform/maxim/maxim_spi.c:93]: 32655 is the target that the user builds for (in this case MAX32655). Would MAX32655 or something similar be more clear?

File: drivers/platform/maxim/spi_extra.h (hunk 4 above)
Comment (CiprianRegus) [reply drivers/platform/maxim/spi_extra.h:43]: Yes, it is.

File: drivers/platform/maxim/irq_extra.h (hunk 5 above)
Comment (CiprianRegus) [reply drivers/platform/maxim/irq_extra.h:42]: They are not related, I split the commits.

**Comment by CiprianRegus** [comment]
- Added copyright header and header guards to `maxim_hal.h`
- Split unrelated changes in different commits.
- Removed extra line.
- Astyled

**Comment by CiprianRegus** [comment]
- Added RTC driver update.

File: drivers/platform/maxim/maxim_spi.c (hunk 3 above)
Comment (amiclaus) [reply drivers/platform/maxim/maxim_spi.c:93]: Got it. I guess it's your call if you want or not to add macros.

File: drivers/platform/maxim/maxim_gpio.c [hunk 6]
```diff
@@ -56,6 +56,21 @@ static struct no_os_callback_desc *gpio_callback[N_PORTS][N_PINS];
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 
+static void set_enable(mxc_gpio_regs_t *regs, uint32_t mask, uint8_t is_enabled)
+{
+#if TARGET_NUM == 32650 || TARGET_NUM == 32665
+	if (is_enabled)
```
Comment (amiclaus) [inline drivers/platform/maxim/maxim_gpio.c:62]: nit: we can use local variable and remove duplicated if else statement.

**Review by amiclaus — APPROVED** [review]
lgtm, one minor comment.

**Comment by CiprianRegus** [comment]
- Added changes to I2C driver.
- Included missing headers in maxim_rtc.c and maxim_i2c.c
- Astyled.
